api4/team_test.go

```diff
@@ -493,4 +517,76 @@ func TestGetMyTeamsUnread(t *testing.T) {
 	Client.Logout()
 	_, resp = Client.GetTeamsUnreadForUser(user.Id, "")
 	CheckUnauthorizedStatus(t, resp)
-}
\ No newline at end of file
+}
+
+func TestUpdateTeamDisplayName(t *testing.T) {
+	th := Setup().InitBasic().InitSystemAdmin()
+	defer TearDown()
+	Client := th.Client
+
+	team := &model.Team{DisplayName: "Name", Name: "z-z-" + model.NewId() + "a", Email: "success+" + model.NewId() + "@simulator.amazonses.com", Type: model.TEAM_OPEN}
+	team, _ = Client.CreateTeam(team)
+
+	team.DisplayName = "Updated Name"
+	uteam, resp := Client.UpdateTeam(team.Id, team);
+	CheckNoError(t, resp)
+
+	if uteam.DisplayName != "Updated Name" {
+		t.Fatal("Update failed")
+	}
+
+	_, resp = Client.UpdateTeam("fake", team)
+	CheckForbiddenStatus(t, resp)
+
+	Client.Logout()
+	_, resp = Client.UpdateTeam(uteam.Id, team)
+	CheckUnauthorizedStatus(t, resp)
+}
+
+func TestUpdateTeamAllowOpenInvite(t *testing.T) {
+	th := Setup().InitBasic().InitSystemAdmin()
+	defer TearDown()
+	Client := th.Client
+
+	team := &model.Team{DisplayName: "Name", AllowOpenInvite: false, Name: "z-z-" + model.NewId() + "a", Email: "success+" + model.NewId() + "@simulator.amazonses.com", Type: model.TEAM_OPEN}
+	team, _ = Client.CreateTeam(team)
+
+	team.AllowOpenInvite = true
+	uteam, resp := Client.UpdateTeam(team.Id, team);
+	CheckNoError(t, resp)
+
+	if uteam.AllowOpenInvite != true {
+		t.Fatal("Update failed")
+	}
+
+	_, resp = Client.UpdateTeam("fake", team)
+	CheckForbiddenStatus(t, resp)
+
+	Client.Logout()
+	_, resp = Client.UpdateTeam(uteam.Id, team)
+	CheckUnauthorizedStatus(t, resp)
+}
+
+func TestUpdateTeamInviteId(t *testing.T) {
+	th := Setup().InitBasic().InitSystemAdmin()
+	defer TearDown()
+	Client := th.Client
+
+	team := &model.Team{DisplayName: "Name", InviteId: "inviteid0", Name: "z-z-" + model.NewId() + "a", Email: "success+" + model.NewId() + "@simulator.amazonses.com", Type: model.TEAM_OPEN}
+	team, _ = Client.CreateTeam(team)
+
+	team.InviteId = "inviteid1"
+	uteam, resp := Client.UpdateTeam(team.Id, team);
+	CheckNoError(t, resp)
+
+	if uteam.InviteId != "inviteid1" {
+		t.Fatal("Update failed")
+	}
+
+	_, resp = Client.UpdateTeam("fake", team)
+	CheckForbiddenStatus(t, resp)
+
+	Client.Logout()
+	_, resp = Client.UpdateTeam(uteam.Id, team)
+	CheckUnauthorizedStatus(t, resp)
+}
```

Comment:
You added a bunch of tests that cover the same endpoint and they have a lot of duplicate code/test cases shared between them. Can you consolidate them all into a single test that covers all the cases?